runtime/lua/vim/lsp.lua

```diff
@@ -1509,24 +1509,28 @@ function lsp.start_client(config)
         { method = method, client_id = client_id, bufnr = bufnr, params = params }
       )
     end, function(request_id)
-      local request = client.requests[request_id]
-      request.type = 'complete'
-      nvim_exec_autocmds('LspRequest', {
-        buffer = bufnr,
-        modeline = false,
-        data = { client_id = client_id, request_id = request_id, request = request },
-      })
-      client.requests[request_id] = nil
+      if vim.fn.bufexists(bufnr) == 1 then
```

Comment:
use `api.nvim_buf_is_valid` and if buffer not valid when callback running can we return `request.type = 'canceld` or other status instead of check many times.

Reply:
instead of check many times just check once like. if buffer not valid request.type is `invalid` then other check is no need. 
```lua
if not  vim.api.nvim_buf_is_valid(bufnr) then
        local request = client.requests[request_id]
        request.type = 'invalid'
        nvim_exec_autocmds('LspRequest', {
          buffer = bufnr,
          modeline = false,
          data = { client_id = client_id, request_id = request_id, request = request },
        })
      end
end
```

Reply:
@glepnir 

Thank you for reviewing! I just replaced `vim.fn.bufexists` with `vim.api.nvim_buf_is_valid` at this time (and tweaked `if` statements to make it avoid only `nvim_exec_autocmds` when the buffer is no longer valid).

> see other people's opinions
👍